### ISSUE TITLE
Emulate real CPU and use the original ROM

### ISSUE THREAD

**Issue by tinue** [TurboGit/ti5x_android#3]
Through a discussion group I found a TI-59 emulator that is running the original ROM of the calculator. The README explains this nicely: [readme](http://www.hrastprogrammer.com/ti59e/index.htm). With this approach, the emulation is - almost by definition - as close to 100% as possible. The ROMs of all of the machines, TI-57, 58, 58c and 59 can be found on the internet; I can share the links if desired.

I was wondering if such an approach has been discussed or even attempted?

**Comment by TurboGit** [comment]
No this has never been attempted. It would be nice indeed but the code is in C/C++ and would have to be rewritten in Java for the Android platform. A project that would require quite some time!

**Comment by tinue** [comment]
It's actually worse than that. As far as I know the author of the TI-59 emulator never published any source code, and does not intend to ever do so ([faq](http://www.hrastprogrammer.com/faq.htm).
Only for the TI-57 is source code of an emulator available, see [pcjs](https://www.pcjs.org/machines/ti/ti57/).

**Comment by mat-c** [comment]
For the record http://bulk.rskey.org/BULK/CALCDOCS/TI/ti-5x_emul.zip emulator source code that use rom code (from https://www.rskey.org/CMS/index.php/the-library/12 )

**Comment by TurboGit** [comment]
@mat-c : Yes I know about this emulator, it is in C though and would require a rewrite in Java or I'll need to learn how to compile this stuff on Android.
